File: lambdas/ride/request.py
import os
import boto3
from uuid import uuid4
import json

TRIP_TABLE = os.getenv('TRIP_TABLE', 'TripsTable')

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table(TRIP_TABLE)

def handler(event, context):
    print(f'Received event: {event}')
    headers = {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'OPTIONS,POST',
        'Access-Control-Allow-Headers': 'Content-Type,Authorization',
         'Access-Control-Allow-Credentials': 'true'
    }

    if event['httpMethod'] == 'OPTIONS':
        # Handle CORS preflight requests
        return {
            'statusCode': 200,
            'headers': headers,
            'body': json.dumps({'message': 'CORS preflight successful'})
        }

    if event['httpMethod'] == 'POST':
        try:
            # Parse the incoming request body
            body = json.loads(event['body'])
            
            # Generate a unique tripId
            trip_id = str(uuid4())
            
            # Construct the item to be added to DynamoDB
            item = {
                'tripId': trip_id,
                'userId': body.get('userId'),
                'pickup': body.get('pickup'),
                'dropoff': body.get('dropoff'),
                'price': body.get('price'),
                'time': body.get('time'),
                'rating': body.get('rating'),
                'driverProfile': body.get('driverProfile'),
            }
            
            # Add the item to DynamoDB
            table.put_item(Item=item)
            
            return {
                'statusCode': 200,
                'headers': headers,
                'body': json.dumps({
                    'message': 'Trip successfully added',
                    'tripId': trip_id,
                    'status': 200,
                })
            }
        except Exception as e:
            print(e)
            return {
                'statusCode': 500,
                'headers': headers,
                'body': json.dumps({'message': 'Error adding trip'})
            }
    else:
        return {
            'statusCode': 405,
            'headers': headers,
            'body': json.dumps({'message': 'Method not allowed'})
        }

File: lambdas/ride/request.ts
import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
import * as AWS from 'aws-sdk';
import { v4 as uuidv4 } from 'uuid';

const TRIP_TABLE = process.env.TRIP_TABLE || 'TripsTable';

const dynamodb = new AWS.DynamoDB.DocumentClient();

export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
    console.log(`Received event: ${JSON.stringify(event)}`);

    const headers = {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'OPTIONS,POST',
        'Access-Control-Allow-Headers': 'Content-Type,Authorization',
        'Access-Control-Allow-Credentials': 'true',
    };

    if (event.httpMethod === 'OPTIONS') {
        // Handle CORS preflight requests
        return {
            statusCode: 200,
            headers,
            body: JSON.stringify({ message: 'CORS preflight successful' }),
        };
    }

    if (event.httpMethod === 'POST') {
        try {
            // Parse the incoming request body
            const body = JSON.parse(event.body || '{}');

            // Generate a unique tripId
            const tripId = uuidv4();

            // Get the current date and time for tripDate
            const tripDate = new Date().toISOString();  // ISO 8601 format

            // Construct the item to be added to DynamoDB
            const item = {
                tripId,
                userId: body.userId,
                pickup: body.pickup,
                dropoff: body.dropoff,
                price: body.price,
                time: body.time,
                rating: body.rating,
                driverProfile: body.driverProfile,
                pickupName: body.pickupName,
                dropoffName: body.dropoffName,
                tripDate,
                pickupCoordinates: body.pickupCoordinates,
                dropoffCoordinates: body.dropoffCoordinates,
            };

            // Add the item to DynamoDB
            await dynamodb.put({
                TableName: TRIP_TABLE,
                Item: item,
            }).promise();

            return {
                statusCode: 200,
                headers,
                body: JSON.stringify({
                    message: 'Trip successfully added',
                    tripId,
                    status: 200,
                }),
            };
        } catch (error) {
            console.error(`Exception occurred: ${error}`);
            return {
                statusCode: 500,
                headers,
                body: JSON.stringify({ message: 'Error adding trip', error: error.message }),
            };
        }
    } else {
        return {
            statusCode: 405,
            headers,
            body: JSON.stringify({ message: 'Method not allowed' }),
        };
    }
};
